resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -308,7 +307,7 @@ private[spark] class ApplicationMaster(
       // The client-mode AM doesn't listen for incoming connections, so report an invalid port.
       registerAM(Utils.localHostName, -1, sparkConf,
         sparkConf.getOption("spark.driver.appUIAddress"), appAttemptId)
-      addAmIpFilter(Some(driverRef), ProxyUriUtils.getPath(appAttemptId.getApplicationId))
+      addAmIpFilter(Some(driverRef), s"/proxy/$appAttemptId")
```

Comment:
Why we are not using `getApplicationId` as before?

Reply:
Oops my bad. Should be `appAttemptId.getApplicationId`.

Reply:
BTW, do we need uri encode as `ProxyUriUtils.getPath` does internally?

Reply:
Yeah we need that too. Good catch. I had it before but something messed up during rebase. Will fix.